apps/job-service/src/jobs/delayed-jobs.controller.ts

```diff
@@ -1,38 +1,139 @@
+  // job, there is no effective way to make a policy for jobs until we expand the service to
+  // include an owner ID on the job table.
   async findOne(@Param('uuid') pathUUID: string): Promise<JsonApiDocument> {
-    const job = await DelayedJob.findOne({ where: { uuid: pathUUID }});
+    const job = await DelayedJob.findOne({ where: { uuid: pathUUID } });
     if (job == null) throw new NotFoundException();
 
-    // Note: Since jobs are very generic and we don't track which resources are related to a given
-    // job, there is no effective way to make a policy for jobs until we expand the service to
-    // include an owner ID on the job table.
-
     return buildJsonApi()
       .addData(pathUUID, new DelayedJobDto(job))
       .document.serialize();
   }
-}
+
+  @Patch('bulk-clear')
+  @ApiOperation({
+    operationId: 'bulkClearJobs',
+    summary: 'Bulk update jobs to modify isAcknowledged for specified job IDs',
+    description: `Accepts a JSON:API-compliant payload to bulk update jobs, allowing each job's isAcknowledged attribute to be set to true or false.`,
+  })
+  @ApiBody({
+    description: 'JSON:API bulk update payload for jobs',
+    type: JobBulkUpdateBodyDto,
+    examples: {
+      example: {
+        value: {
+          data: [
+            {
+              type: 'jobs',
+              uuid: 'uuid-1',
+              attributes: {
+                isAcknowledged: true,
+              },
+            },
+            {
+              type: 'jobs',
+              uuid: 'uuid-2',
+              attributes: {
+                isAcknowledged: false,
+              },
+            },
+          ],
+        },
+      },
+    },
+  })
+  @ApiException(() => UnauthorizedException, { description: 'Authentication failed.' })
+  @ApiException(() => BadRequestException, { description: 'Invalid payload or IDs provided.' })
+  @ApiException(() => NotFoundException, { description: 'One or more jobs specified in the payload could not be found.' })
+  async bulkClearJobs(
+    @Body() bulkClearJobsDto: JobBulkUpdateBodyDto,
+    @Request() { authenticatedUserId }
+  ): Promise<{ data: JobData[] }> {
+    const jobUpdates = bulkClearJobsDto.data;
+
+    if (!jobUpdates || jobUpdates.length === 0) {
+      throw new BadRequestException('No jobs provided in the payload.');
+    }
+
+    const updatePromises = jobUpdates.map(async (job) => {
+      const [updatedCount] = await DelayedJob.update(
+        { isAcknowledged: job.attributes.isAcknowledged },
+        {
+          where: {
+            uuid: job.uuid,
+            createdBy: authenticatedUserId,
+            status: { [Op.ne]: 'pending' },
+          },
+        }
+      );
+
+      if (updatedCount === 0) {
+        throw new NotFoundException(`Job with UUID ${job.uuid} could not be updated.`);
+      }
+
+      return job;
+    });
+
+    const updatedJobs = await Promise.all(updatePromises);
+
+    return {
+      data: updatedJobs,
+    };
```

Comment:
Please always use the JsonApiBuilder pattern for responses. You can also either remove the return type from the method, or if you want to keep it, change it to `Promise<JsonApiDocument>`

Reply:
Changed to correct pattern